### ISSUE TITLE
WiFi Rev2 RGB control

### ISSUE THREAD

**Issue by sabas1080** [arduino-libraries/ArduinoBLE#3]
Please add API to control the built-in RGB which is connected to pins 25,26,27 for Arduino MKR and Arduino WiFi Rev2

Thanks

**Comment by facchinm** [comment]
Hi @sabas1080 , do you need to control the RGB LED from BLE? If using WiFI, please refer to  https://github.com/arduino-libraries/WiFiNINA/issues/24 for instructions. 
BLE is a bit different since we need a side channel to communicate with ESP32 internally (all the "normal" communication is redirected to BLE "core"). @sandeepmistry any idea on how we could solve it?

**Comment by sabas1080** [comment]
Thanks @facchinm

**Comment by sandeepmistry** [comment]
Hi @sabas1080,

I think we are going to hold off on implementing something for a while.

Currently, when the NINA is in Bluetooth mode, the ESP32 firmware puts the module in HCI passthrough mode using the standard UART over HCI protocol.

On the MKR WiFi 1010 the ACK and GPIO0 pins and be used for communications but we can't have SPI communications, and on the Uno WiFi rev2 wth CS pin is free. We could consider to have a serial protocol (TX only) on one of the pins to send RGB values. However, then this should be left out of this library and it would create some divergence for the way things would work when in WiFi mode.

**Comment by sabas1080** [comment]
Hi @sandeepmistry 

ok, i am trying with WifiNINA but not working led RGB, my sketch is

https://github.com/sabas1080/KanoWandHack/blob/master/arduino-wifi-rev2/arduino-wifi-rev2.ino#L132

**Comment by sandeepmistry** [comment]
I suggest trying to add a WiFi.status() call before:

```
  WiFi.status(); <-- added

  WiFiDrv::pinMode(25, OUTPUT);  //GREEN
  WiFiDrv::pinMode(26, OUTPUT);  //RED
  WiFiDrv::pinMode(27, OUTPUT);  //BLUE
  WiFiDrv::analogWrite(25, 128);  // for configurable brightness
  WiFiDrv::analogWrite(26, 128);  // for configurable brightness
  WiFiDrv::analogWrite(27, 128);  // for configurable brightness
```

Once you enable BLE, there is no way to control the LEDs and they will be turn off until WiFi mode is enabled, as ArduinoBLE and WiFiNINA libraries are not compatible.

**Comment by sabas1080** [comment]
then I can not use BLE and WIFI at the same time?

**Comment by sandeepmistry** [comment]
> then I can not use BLE and WIFI at the same time?

Correct, as stated in the forum: http://forum.arduino.cc/index.php?topic=579469.0

>  Please note, WiFi functionality (using the WiFiNINA library) cannot be used at the same time as BLE.

**Comment by sabas1080** [comment]
ok, I hope later you can control, thanks
